bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/StateDescriptionFragmentBuilder.java

```diff
@@ -0,0 +1,179 @@
+            private @Nullable final BigDecimal step = StateDescriptionFragmentBuilder.this.step;
+            private @Nullable final String pattern = StateDescriptionFragmentBuilder.this.pattern;
+            private @Nullable final Boolean readOnly = StateDescriptionFragmentBuilder.this.readOnly;
+            private @Nullable final List<StateOption> options = StateDescriptionFragmentBuilder.this.options;
+
+            @Override
+            public @Nullable BigDecimal getMinimum() {
+                return minimum;
+            }
+
+            @Override
+            public @Nullable BigDecimal getMaximum() {
+                return maximum;
+            }
+
+            @Override
+            public @Nullable BigDecimal getStep() {
+                return step;
+            }
+
+            @Override
+            public @Nullable String getPattern() {
+                return pattern;
+            }
+
+            @Override
+            public @Nullable Boolean isReadOnly() {
+                return readOnly;
+            }
+
+            @Override
+            public @Nullable List<StateOption> getOptions() {
+                return options;
+            }
+
+            @Override
+            public @Nullable StateDescription toStateDescription() {
+                if (minimum == null && maximum == null && step == null && pattern == null && options == null) {
+                    return null;
+                }
+                final Boolean ro = readOnly;
+                return new StateDescription(minimum, maximum, step, pattern, ro == null ? false : ro.booleanValue(),
+                        options);
+            }
+
+        };
+    }
+
+    public StateDescriptionFragmentBuilder withMaximum(BigDecimal maximum) {
+        this.maximum = maximum;
+        return this;
+    }
+
+    public StateDescriptionFragmentBuilder withMinimum(BigDecimal minimum) {
+        this.minimum = minimum;
+        return this;
+    }
+
+    public StateDescriptionFragmentBuilder withStep(BigDecimal step) {
+        this.step = step;
+        return this;
+    }
+
+    public StateDescriptionFragmentBuilder withPattern(String pattern) {
+        this.pattern = pattern;
+        return this;
+    }
+
+    public StateDescriptionFragmentBuilder withReadOnly(Boolean readOnly) {
+        this.readOnly = readOnly;
+        return this;
+    }
+
+    public StateDescriptionFragmentBuilder withOptions(List<StateOption> options) {
+        this.options = options;
+        return this;
+    }
+
+    /**
+     * Set the builder fields from all non-null fields of the given {@link StateDescriptionFragment}.
+     *
+     * @param fragment a {@link StateDescriptionFragment} this builder should incorporate.
+     * @return the builder.
+     */
+    public StateDescriptionFragmentBuilder withStateDescriptionFragment(StateDescriptionFragment fragment) {
+        if (fragment.getMinimum() != null) {
```

Comment:
This logic seems to be wrong: The builder is supposed to merge in the given StateDescriptionFragment. So it should not override its internal fields if already set. It should only take values for still unknown values.

Reply:
What if I have a state description (let‘s say min, max, readonly) and want to change readonly, but leave the others untouched. 
As far as I understood I should create a fragment with all fields null, except the readonly field. If already set fields are not overridden, this change will not take effect. Is that intended?

Reply:
@J-N-K I would expect:

```Java
StateDescriptionFragmentBuilder builder = StateDescriptionFragmentBuilder.instance().withStateDescription(stateDescription).withReadOnly(newReadOnlyValue);
```

Reply:
Yes, correct.
With the latest code and behaviour as described above I did refactor the `withStateDescription...` methods to `mergeStateDescription...` and also adopted the JavaDoc. I hope with this it is clear that `with...` does override but `merge...` does not.